src/utils/queryBuilder.ts

```diff
@@ -52,24 +52,41 @@ const _generateRangeBlock = (fields: IDateValues): estypes.QueryDslQueryContaine
     true,
   );
 
+
   const rangeBlock: estypes.QueryDslQueryContainer[] = [
     {
       bool: {
         should: [
           {
-            range: {
-              'resourceTemporalExtentDetails.start.date': {
-                gte: startDateValue,
-                lte: toDateValue,
-              },
+            bool: {
+              must: [
+                { range: { 'resourceTemporalExtentDetails.start.date': { lt: startDateValue } } },
+                { range: { 'resourceTemporalExtentDetails.end.date': { gt: toDateValue } } },
+              ],
+            },
+          },
+          {
+            bool: {
+              must: [
+                { range: { 'resourceTemporalExtentDetails.start.date': { lt: startDateValue } } },
+                { range: { 'resourceTemporalExtentDetails.end.date': { lt: toDateValue } } },
+              ],
+            },
+          },
+          {
+            bool: {
+              must: [
+                { range: { 'resourceTemporalExtentDetails.start.date': { gt: startDateValue } } },
+                { range: { 'resourceTemporalExtentDetails.end.date': { lt: toDateValue } } },
+              ],
```

Comment:
As we discussed this query needs modification to cover all the scenarios specified on the AC. Please change the query accordingly

Reply:
Resolved